src/Metalama.Patterns.NotifyPropertyChanged/Implementation/Natural/DependencyGraphNode.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) SharpCrafters s.r.o. See the LICENSE.md file in the root directory of this repository root for details.
+
+using Metalama.Framework.Aspects;
+using Metalama.Framework.Code;
+using System.Text;
+
+namespace Metalama.Patterns.NotifyPropertyChanged.Implementation.Natural;
+
+/// <summary>
+/// Dependency graph node specialized for the current implementation strategy of <see cref="NotifyPropertyChangedAttribute"/>.
+/// </summary>
+[CompileTime]
+internal sealed class DependencyGraphNode : DependencyGraph.Node<DependencyGraphNode>
```

Comment:
There is a bad separation of concerns in this class.
1. This logic should not be derived from DependencyGraph.Node.
2. It is probably related to BuildAspectContext (refactored as an builder+immutable pattern)
3. This class itself could/should be made immutable.